### PR TITLE
Improve code coverage for cunumeric.bincount

### PR DESCRIPTION
**Pull request by yimoj**
For the three tests that are shown as removed, they're just moved into `TestBincountNegative`.

### CODE REVIEW

File: tests/integration/test_bincount.py [hunk 1]
```diff
@@ -27,36 +27,92 @@
...
+    def test_dtype_negative(self):
+        expected_exc = TypeError
+        arr_np = np.arange(5, dtype=float)
+        arr_num = num.arange(5, dtype=float)
+        with pytest.raises(expected_exc):
+            np.bincount(arr_np)
+        with pytest.raises(expected_exc):
+            num.bincount(arr_num)
+
+    def test_array_negateive(self):
```
Comment (bryevdv) [inline tests/integration/test_bincount.py:40]: typo
```suggestion
    def test_array_negative(self):
```

File: tests/integration/test_bincount.py [hunk 2]
```diff
@@ -105,6 +161,23 @@ def test_bincount_weights_high_bins(dtype):
...
+        pytest.param(np.array((22,)), marks=pytest.mark.xfail),
+    ],
+    ids=str,
+)
+def test_bincount_size_one(weights):
+    arr_np = np.random.randint(0, 255, size=1)
+    arr_num = num.array(arr_np)
+    bins_np = np.bincount(arr_np, weights=weights)
+    bins_num = num.bincount(arr_num, weights=weights)
+    assert np.array_equal(bins_np, bins_num)
```
Comment (bryevdv) [inline tests/integration/test_bincount.py:178]: Can you elaborate on the failures here, e.g. there are actual differing value from the libraries?

**Review by bryevdv — APPROVED** [review]
One typo to fix otherwise LGTM

File: tests/integration/test_bincount.py [hunk 3]
```diff
@@ -105,6 +161,23 @@ def test_bincount_weights_high_bins(dtype):
...
+    "weights",
+    [
+        None,
+        pytest.param(np.array((0.5,)), marks=pytest.mark.xfail),
+        pytest.param(np.array((22,)), marks=pytest.mark.xfail),
+    ],
+    ids=str,
+)
+def test_bincount_size_one(weights):
+    arr_np = np.random.randint(0, 255, size=1)
```
Comment (manopapad) [inline tests/integration/test_bincount.py:174]: I confirmed that this is a bug in the code. Can you please try applying this patch and confirm that the singleton test with weight now pass?

```
diff --git a/cunumeric/module.py b/cunumeric/module.py
index cfe0a7b4..a5955dfb 100644
--- a/cunumeric/module.py
+++ b/cunumeric/module.py
@@ -6361,7 +6361,7 @@ def bincount(
         else:
             out = zeros((minlength,), dtype=weights.dtype)
             index = x[0]
-            out[index] = weights[index]
+            out[index] = weights[0]
     else:
         # Normal case of bincount
         if weights is None:
```

Comment (yimoj) [reply tests/integration/test_bincount.py:174]: It passes after applying the patch.

Comment (bryevdv) [reply tests/integration/test_bincount.py:184]: @yimoj can you apply the patch and update the test in this PR? For something like this it's easier to have fewer PRs and also less chance it falls through the cracks. If you want an issue to to track for the bug one can be made that also points to this PR.

Comment (yimoj) [reply tests/integration/test_bincount.py:174]: @bryevdv sure and done.

**Comment by yimoj** [comment]
The last force push is just to re-run the CI check that appeared to be stuck.

**Pull request merged**